### PR TITLE
Refactor assessments view

### PR DESCRIPTION
**Pull request by tuesmiddt**
Fixes #116

### CODE REVIEW

File: lib/cadet_web/views/view_helpers.ex [hunk 1]
```diff
@@ -6,4 +6,77 @@ defmodule CadetWeb.ViewHelpers do
...
+    barbar: "ha",
+    foobar: "hoha"
+  }
+
+  field_list = [:foofoo, :barbar]
+
+  transform_map_for_view(source, field_list)
+  > %{
+    foofoo: "ho",
+    barbar: "bar"
```
Comment (indocomsoft) [inline lib/cadet_web/views/view_helpers.ex]: Did you mean `barbar: "ha"` lol

File: lib/cadet_web/views/view_helpers.ex [hunk 2]
```diff
@@ -6,4 +6,77 @@ defmodule CadetWeb.ViewHelpers do
...
+  }
+
+  transform_map_for_view(source, key_transformations)
+  > %{
+    foo: Map.get(source, :foofoo),
+    bar: Map.get(source, :barbar)
+  }
+
+  function_transformations = %{
+    foo: fn source -> source.foofoo <> "hoho",
```
Comment (indocomsoft) [inline lib/cadet_web/views/view_helpers.ex]: I think this is a better design:
```elixir
source = %{
  foofoo: "ho",
  barbar: "ha",
  foobar: "hoha"
}

function_transformations = %{
  foofoo: &(&1 <> "tst")
}

transform_map_for_view(source, functions_transformations)
# > %{
#   foofoo: "hotst"
# }
```

Not really sure why you decided to do what you did. Maybe you could elaborate?

Comment (tuesmiddt) [reply lib/cadet_web/views/view_helpers.ex]: For me, the primary intention is to create a standardised way to transform from a `source_map` into a `view_map`. Other than the direct `source_field_name` to `view_field_name` transformation, I wanted to be able to define a `view_field` as a `fn source_map -> my_custom_transformation_to_value(source_map) end`. The transformation could involve aggregation of multiple fields from the `source_map`, calculations, etc. 

Your suggestion only applies to when `source_map` and `view_map` have the same `field_name` and you are doing a direct `value` -> `new_value` transformation.

Comment (indocomsoft) [reply lib/cadet_web/views/view_helpers.ex]: Ah ok fair enough.

**Pull request merged**